site/viamguides/src/environment-variables/environment-variables.md

```diff
@@ -0,0 +1,223 @@
+[Set up a Python virtual environment](https://docs.viam.com/sdks/python/python-venv/) to avoid any conflicts with other projects or your system.
+1. From the command line in your terminal window, install [`virtualenv` using `pip`](https://virtualenv.pypa.io/en/latest/installation.html#via-pip)
+    ```bash
+    python3 -m pip install --user virtualenv
+    ```
+1. **Create and activate a virtual environment**: In the project directory `viam-rover`, create and activate a virtual environment called `.venv` for Python to run in.
+    ```bash
+    python3 -m venv .venv
+    source .venv/bin/activate
+    ```
+Now, `.venv` prepends the commands in your terminal window to indicate the Python packages being used are from this particular environment. You can exit this environment by running `deactivate`.
+
+### Install the Python SDK
+
+Install the Viam Python SDK and all required general dependencies.
+
+1. **Install the SDK**: From the command line in your terminal window (inside the activated `.venv` Python environment), install the Viam Python SDK:
+    ```bash
+    pip3 install viam-sdk
+    ```
+
+    > aside negative
+    > Windows is not supported. If you are using Windows, use the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) and install the Python SDK using the preceding instructions for Linux. For other unsupported systems, see [Installing from source](https://python.viam.dev/#installing-from-source).
+
+<!-- ------------------------ -->
+## Using variables in local development
+Duration: 5
+
+### Copy the sample code
+1. **Refer to Python sample code**: In the Viam app, go to the **CONNECT** tab, and select **Python**. 
+  ![Connect tab](assets/Connect.png)
+1. **Show secrets**: The sample code shows you how to authenticate and connect to a machine, as well as some of the methods you can use on your configured components and services. To show your machine’s API key and API key ID in the sample code, toggle **Include secret** on the CONNECT tab’s Code sample page. 
+  ![Code sample](assets/Code.png)
+1. **Hide secrets**: Instead of leaving our sensitive credentials hard coded in this code sample, let's set up our API key and API key ID as environment variables. Toggle **Include secrets** once again, this time to remove the credentials from the code sample. 
+1. **Copy the code**: From the command line in your terminal window, create a new file called `square.py`, and copy the code sample into this file.
+   ```bash
+   touch square.py
+   ```
+### Run the sample code
+1. **Create `.env` file**: From the command line, create a new file called `.env` to store our environment variables to use in local development.
+    ```bash
+    touch .env
+    ```
+1. **Input environment variables**: Add your environment variables to the `.env` file, formatted like the following:
+    ```
+    ROBOT_API_KEY="sijdb24bjnxsmp18mij0hace6smihu0r"
+    ROBOT_API_KEY_ID="9e135013-f3bf-40fc-a0e3-e6fc66a418d7"
```

Comment:
Should we call out to fill in the values for API-KEY and API-KEY-ID from the code sample as these environment variables instead of copying this explicitly?